### ISSUE TITLE
Update topic on Adding new diagnostic messages

### ISSUE THREAD

**Issue by lionelmoi** [dita-ot/docs#156]
The [Adding new diagnostic messages](http://www.dita-ot.org/dev/dev_ref/plugin-messages.html) topic seems outdated.

It mentions the `msgnum` and `msgsev` parameters which the code says are deprecated since 2.3, replaced by the `id` parameter.

It may also be lacking some information on how to use the output-message template in a plugin.

**Comment by infotexture** [comment]
@lionelmoi Are you able to submit a pull request with proposed changes?

**Comment by lionelmoi** [comment]
@infotexture yes that is the idea 😄 

I will need some input regarding behavior of the toolkit for FATAL errors as I do not yet understand how they work.

**Comment by lionelmoi** [comment]
@jelovirt 
Here are my questions.

- What is the pattern for the message id prefix? 
  - Is there one? 
  - Does it have to be DOTX?

- When message type to F it triggers a message with @terminate set to "yes". How do you create a message that crashes the toolkit?

In `plugin:org.dita.base:xsl/common/output-message.xsl`

```
      <xsl:when test="$msgsev = 'F'">
        <xsl:message terminate="yes">
          <xsl:value-of select="$m" separator=""/>
        </xsl:message>
      </xsl:when>
```
**output-message template parameters**
- `$ctx`
  - when to use it? When calling the template in a function?

- `$msg`
  - what is the syntax to use params in the message? Should they be in quotation marks?

- `$msgcat`
  - it is deprecated but I think it creates an issue when $msgprefix is not set. Does it mean $msg prefix should always be set up.

- `$id`
  - should the pattern always follow the DOTX\d\d\d(I|W|E|F)?

**Comment by jelovirt** [comment]
The prefix by convention is 4 character prefix, 3 number message number, 1 character severity indicator, but it can be anything plugin developer wants.

By convention fatal messages will fail the whole conversion. Just throw a fatal error message in XSLT; in Java throw an exception.

**Comment by infotexture** [comment]
Implemented by @lionelmoi in #170.
